File: src/main/java/codes/nibby/yi/io/GameParseException.java
package codes.nibby.yi.io;

public class GameParseException extends Exception {

    public GameParseException() {
        super();
    }

    public GameParseException(String message) {
        super(message);
    }

    public GameParseException(String message, Throwable cause) {
        super(message, cause);
    }

    public GameParseException(Throwable cause) {
        super(cause);
    }

    protected GameParseException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: yi.core/src/main/kotlin/yi/core/go/GameParseException.kt
package yi.core.go

/**
 * Thrown when an error is encountered while parsing documents as [GameModel].
 * Usually caused by malformed game data.
 */
class GameParseException(message: String, cause: Throwable?) : Exception(message, cause) {

    constructor(message: String) : this(message, null)

}